src/immer/create-slice-with-immer.ts

```diff
@@ -0,0 +1,46 @@
+import { produce } from 'immer';
+
+// Utility type to infer the argument types of the actions
+type InferArgs<T> = T extends (...args: infer A) => void ? A : never;
+
+export type SliceActions<Value> = {
+  [actionName: string]: (...args: never[]) => (prev: Value) => void;
+};
+
+export type SliceConfig<
+  Name extends string,
+  Value,
+  Actions extends SliceActions<Value>,
+> = {
+  name: Name;
+  value: Value;
+  actions: Actions;
+};
+
+type ImmerActions<Value, Actions extends SliceActions<Value>> = {
+  [K in keyof Actions]: (...args: InferArgs<Actions[K]>) => (prev: Value) => void;
```

Comment:
Hm? Should it return `Value` instead of `void`?

Reply:
Functions wrapped with produce modify the draft state directly and do not need to return anything, the official documentation states that recipe functions perform mutations without returning a new object​ ([ImmerJS](https://immerjs.github.io/immer/return/))​.

Reply:
If I understand correctly, `ImmerActions` is what is after `produce` is applied.
What `produce` takes is `SliceActions`, no?